DS13/code/modules/overmap/warpcore.dm

```diff
@@ -0,0 +1,676 @@
+	else
+		core.AI = src
+
+/obj/item/dilithium_matrix //An interface for a dilithium crystal. It will take damage in the core based on how hot the reaction is. When it starts to get out of alignment you have to take it out and realign it with a safe cracking minigame.
+	name = "Dilithium matrix"
+	desc = "A device designed to harness the power of a dilithium crystal for interstellar travel. It needs occasional realignment"
+	icon = 'DS13/icons/obj/power/warpcore/machines.dmi'
+	icon_state = "matrix"
+	var/alignment = 100 //this is damaged per tick based on how well youve aligned it. If this reaches 0 youre looking at a core breach as it regulates  the matter annihilation reaction
+	var/max_alignment = 100
+	var/obj/item/dilithium/stored
+	var/target_freq = 1
+	var/freq = 5
+	var/combo_target = "omega" //Randomized sequence for the recalibration minigame.
+	var/list/letters = list("delta,", "omega,", "phi,")
+	var/combo = null
+	var/combocount = 0 //How far into the combo are they?
+
+/obj/item/dilithium_matrix/examine(mob/user)
+	. = ..()
+	to_chat(user, "<span class='warning'>[src]'s realignment sequence is: [combo_target].</span>")
+
+/obj/item/dilithium_matrix/Initialize()
+	. = ..()
+	combo_target = "[pick(letters)][pick(letters)][pick(letters)][pick(letters)][pick(letters)]"
+
+/obj/item/dilithium_matrix/proc/absorb_damage(var/amount)
+	alignment -= amount
+	if(alignment > 0)
+		return TRUE
+	if(alignment <= 0)
+		qdel(stored)
+		stored = null
+		alignment = 0
+		icon_state = "matrix"
+		visible_message("<span class='warning'>[src]'s dilithium crystal shatters!</span>")
+		return FALSE
+
+/obj/item/dilithium_matrix/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/dilithium))
+		to_chat(user, "<span class='notice'>You slot [W] into [src].</span>")
+		W.forceMove(src)
+		stored = W
+		icon_state = "matrix-dilithium"
+		playsound(src, 'DS13/sound/effects/warpcore/loader_step1.ogg', 50, 1)
+
+/obj/item/dilithium_matrix/attack_self(mob/user)
+	var/sound/thesound = pick(GLOB.bleeps)
+	SEND_SOUND(user, thesound)
+	var/list/options = letters
+	for(var/option in options)
+		options[option] = image(icon = 'DS13/icons/actions/engine_actions.dmi', icon_state = "[option]")
+	var/dowhat = show_radial_menu(user,src,options)
+	if(!dowhat)
+		return
+	combo += "[dowhat]"
+	combocount ++
+	to_chat(user, "<span class='warning'>You inputted [dowhat] into the command sequence.</span>")
+	playsound(src, 'sound/machines/sm/supermatter3.ogg', 20, 1)
+	if(combocount <= 4)
+		addtimer(CALLBACK(src, .proc/attack_self, user), 2)
+	if(combocount >= 5) //Completed the sequence
+		if(combo == combo_target)
+			to_chat(user, "<span class='warning'>Realignment of dilithium matrix complete.</span>")
+			playsound(src, 'sound/machines/sm/supermatter1.ogg', 30, 1)
+			alignment = max_alignment
+		else
+			to_chat(user, "<span class='warning'>Realignment failed continued failure risks damage to dilithium crystal sample. Rotating command sequence.</span>")
```

Comment:
Missing punctuation between "Realignment failed" and "continued failure risks damage to dilithium crystal sample"